sharding-orchestration/sharding-orchestration-core/src/main/java/org/apache/shardingsphere/orchestration/internal/registry/config/service/ConfigurationService.java

```diff
@@ -174,7 +174,13 @@ private boolean hasProperties() {
      * @return is sharding rule or not
      */
     public boolean isShardingRule(final String shardingSchemaName) {
-        return regCenter.getDirectly(configNode.getRulePath(shardingSchemaName)).contains("tables:\n");
+        if (regCenter.getDirectly(configNode.getRulePath(shardingSchemaName)).contains("encryptRule:\n")) {
```

Comment:
Incorrect if sharding only

Reply:
The rule configuration in registry center  format is as follows  when sharding only.
```
tables:
  t_address:
    actualDataNodes: ds_${0..1}.t_address
    logicTable: t_address
  t_order:
    actualDataNodes: ds_${0..1}.t_order_${0..1}
    databaseStrategy:
      inline:
        algorithmExpression: ds_${user_id % 2}
        shardingColumn: user_id
    keyGenerator:
      column: order_id
      type: SNOWFLAKE
    logicTable: t_order
    tableStrategy:
      inline:
        algorithmExpression: t_order_${order_id % 2}
        shardingColumn: order_id
  t_order_item:
    actualDataNodes: ds_${0..1}.t_order_item_${0..1}
    databaseStrategy:
      inline:
        algorithmExpression: ds_${user_id % 2}
        shardingColumn: user_id
    keyGenerator:
      column: order_id
      type: SNOWFLAKE
    logicTable: t_order_item
    tableStrategy:
      inline:
        algorithmExpression: t_order_item_${order_id % 2}
        shardingColumn: order_id
```

We  just make sure  the rule is not  encrypt  only. the encrypt only  rule configuration in registry center  is as follows.
```
encryptors:
    encryptor_aes:
      type: aes
      props:
        aes.key.value: 123456abc
    encryptor_md5:
      type: md5
 tables:
    t_user:
      columns:
        user_name:
          plainColumn: user_name
          cipherColumn: user_name_plain
          encryptor: encryptor_aes
```

The complex rule is as follows. it will be as  shardingRule
  ```
tables:
    t_user:
      actualDataNodes: ds_${0..1}.t_user_${0..1}
      databaseStrategy:
        inline:
          shardingColumn: user_id
          algorithmExpression: ds_${user_id % 2}
      tableStrategy:
        inline:
          shardingColumn: user_id
          algorithmExpression: t_user_${user_id % 2}
      keyGenerator:
        type: SNOWFLAKE
        column: user_id
  defaultTableStrategy:
    none:
      
  defaultDatabaseStrategy:
    none: 
    
  masterSlaveRules:
    ds_0:
      masterDataSourceName: ds_master_0
      slaveDataSourceNames: [ds_master_0_slave_0, ds_master_0_slave_1]
      loadBalanceAlgorithmType: ROUND_ROBIN
    ds_1:
      masterDataSourceName: ds_master_1
      slaveDataSourceNames: [ds_master_1_slave_0, ds_master_1_slave_1]
      loadBalanceAlgorithmType: ROUND_ROBIN
      
  encryptRule:
    encryptors:
      encryptor_aes:
        type: aes
        props:
          aes.key.value: 123456abc
    tables:
      t_user:
        columns:
          user_name:
            plainColumn: user_name_plain
            cipherColumn: user_name
            encryptor: encryptor_aes
```

Reply:
Here, In the judgement of sharding rule , we just make sure it's not encrypt only. All the complex  scenes are sharding.

Reply:
The difference of Rule Configuration in registry center  in  encrypt only and cmplex scene is as follows:

 **encrypt only**
```
encryptors:
    encryptor_aes:
      type: aes
      props:
        aes.key.value: 123456abc
    encryptor_md5:
      type: md5
tables:
    t_user:
      columns:
        user_name:
          plainColumn: user_name_plain
          cipherColumn: user_name
          encryptor: encryptor_md5
```
**complex scene**
```
  tables:
    t_user:
      actualDataNodes: ds_${0..1}.t_user_${0..1}
      databaseStrategy:
        inline:
          shardingColumn: user_id
          algorithmExpression: ds_${user_id % 2}
      tableStrategy:
        inline:
          shardingColumn: user_id
          algorithmExpression: t_user_${user_id % 2}
      keyGenerator:
        type: SNOWFLAKE
        column: user_id
  defaultTableStrategy:
    none:
      
  defaultDatabaseStrategy:
    inline:
      shardingColumn: user_id
      algorithmExpression: ds_${user_id % 2}
  masterSlaveRules:
    ds_0:
      masterDataSourceName: ds_master_0
      slaveDataSourceNames: [ds_master_0_slave_0, ds_master_0_slave_1]
      loadBalanceAlgorithmType: ROUND_ROBIN
    ds_1:
      masterDataSourceName: ds_master_1
      slaveDataSourceNames: [ds_master_1_slave_0, ds_master_1_slave_1]
      loadBalanceAlgorithmType: ROUND_ROBIN
      
  encryptRule:
    encryptors:
      encryptor_aes:
        type: aes
        props:
          aes.key.value: 123456abc
    tables:
      t_user:
        columns:
          user_name:
            plainColumn: user_name_plain
            cipherColumn: user_name
            encryptor: encryptor_aes
```